File: Backend-Java/src/main/java/com/example/demo/entity/Student.java
package com.example.demo.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;

@Entity
@Table(name = "students")
@Getter
@Setter
public class Student {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "student_id")
    private Integer studentId;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "inquiry_id")
    private Inquiry inquiry;

    @Column(name = "first_name", nullable = false, length = 150)
    private String firstName;

    @Column(name = "last_name", nullable = false, length = 150)
    private String lastName;

    @Column(name = "parent_name", nullable = false, length = 150)
    private String parentName;

    @Column(name = "parent_phone", length = 15)
    private String parentPhone;

    @Column(name = "email", unique = true, length = 150)
    private String email;

    @Column(name = "phone", length = 15)
    private String phone;

    @Column(name = "dob")
    private LocalDate dob;

    @Enumerated(EnumType.STRING)
    @Column(name = "gender", length = 10)
    private Gender gender;

    @Column(name = "address_line1", length = 255)
    private String addressLine1;

    @Column(name = "address_line2", length = 255)
    private String addressLine2;

    @Column(name = "city", length = 100)
    private String city;

    @Column(name = "state", length = 100)
    private String state;

    @Column(name = "pincode", length = 10)
    private String pincode;

    @Column(name = "photo_url", length = 500)
    private String photoUrl;

    @Column(name = "qualification", length = 150)
    private String qualification;

    @Column(name = "reg_date", nullable = false)
    private LocalDate regDate = LocalDate.now();

    public enum Gender {
        Male,
        Female,
        Other
    }
}

File: dotnet_backend/backend/Models/Student.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ComputerSeekho.Models;

/// <summary>
/// A registered student.
///
/// InquiryId is NOT NULL, which is the database half of BRD section 4.1 —
/// "no enquiry, no registration". Every student is traceable to where they
/// came from, and the schema refuses to hold one that isn't.
/// </summary>
[Table("students")]
public class Student
{
    [Key]
    [Column("student_id")]
    public int StudentId { get; set; }

    [Column("inquiry_id")]
    public int InquiryId { get; set; }

    [Column("first_name")]
    [Required, MaxLength(150)]
    public string FirstName { get; set; } = string.Empty;

    [Column("last_name")]
    [Required, MaxLength(150)]
    public string LastName { get; set; } = string.Empty;

    [Column("parent_name")]
    [Required, MaxLength(150)]
    public string ParentName { get; set; } = string.Empty;

    [Column("parent_phone")]
    [MaxLength(15)]
    public string? ParentPhone { get; set; }

    [Column("email")]
    [MaxLength(150)]
    public string? Email { get; set; }

    [Column("phone")]
    [MaxLength(15)]
    public string? Phone { get; set; }

    [Column("dob")]
    public DateOnly? Dob { get; set; }

    [Column("gender")]
    public Gender? Gender { get; set; }

    [Column("address_line1")]
    [MaxLength(255)]
    public string? AddressLine1 { get; set; }

    [Column("address_line2")]
    [MaxLength(255)]
    public string? AddressLine2 { get; set; }

    [Column("city")]
    [MaxLength(100)]
    public string? City { get; set; }

    [Column("state")]
    [MaxLength(100)]
    public string? State { get; set; }

    [Column("pincode")]
    [MaxLength(10)]
    public string? Pincode { get; set; }

    [Column("photo_url")]
    [MaxLength(500)]
    public string? PhotoUrl { get; set; }

    [Column("qualification")]
    [MaxLength(150)]
    public string? Qualification { get; set; }

    [Column("reg_date")]
    public DateOnly RegDate { get; set; } = DateOnly.FromDateTime(DateTime.Today);

    [ForeignKey(nameof(InquiryId))]
    public Inquiry? Inquiry { get; set; }
}
